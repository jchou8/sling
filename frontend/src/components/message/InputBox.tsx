import * as React from 'react';
import './styles.scss';
import {
    TextField,
    IconButton,
    InputAdornment
} from '@material-ui/core'
import Send from '@material-ui/icons/Send'

type InputBoxProps = {
    sendMessage: Function
    enabled: boolean
}

type InputBoxState = {
    input: string
}

class InputBox extends React.Component<InputBoxProps, InputBoxState> {
    constructor(props: InputBoxProps) {
        super(props);
        this.state = {
            input: "",
        }
    }
    handleInputChange = (value: string) => {
        this.setState({ input: value });
    }

    handleSubmit = (e: React.FormEvent) => {
        e.preventDefault();
        this.props.sendMessage(this.state.input)
        this.setState({
            input: "",
        });
    }

    render() {
        return (
            <div className="InputBox">
                <form className="input-form" onSubmit={this.handleSubmit}>
                    <TextField
                        className="input-box"
                        id="input-box"
                        value={this.state.input}
                        onChange={e => this.handleInputChange(e.currentTarget.value)}
                        margin="dense"
                        fullWidth
                        variant="filled"
                        disabled={!this.props.enabled}
                        InputProps={{
                            endAdornment: (
                                <InputAdornment position="end">
                                    <IconButton
                                        id="input-submit"
                                        className="input-submit"
                                        type="submit"
                                        value="Submit"
                                        disabled={!this.props.enabled}
                                    >
                                        <Send />
                                    </IconButton>
                                </InputAdornment>
                            )
                        }}
                    />
                </form>
            </div>
        );
    }
}

export default InputBox;