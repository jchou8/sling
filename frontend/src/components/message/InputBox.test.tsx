import * as React from 'react'
import { createShallow, createMount } from '@material-ui/core/test-utils'
import InputBox from './InputBox'

const sendFn = jest.fn()

describe('<InputBox />', () => {
  let mount: Function
  beforeAll(() => {
    mount = createMount()
  })

  it('should render when enabled', () => {
    const component = mount(<InputBox enabled={true} sendMessage={sendFn} />)
    expect(component).toMatchSnapshot()
  })

  it('should render when disabled', () => {
    const component = mount(<InputBox enabled={false} sendMessage={sendFn} />)
    expect(component).toMatchSnapshot()
  })

  it('should send a message upon submitting', () => {
    const component = mount(<InputBox enabled={true} sendMessage={sendFn} />)
    component.setState({ input: 'hello' })
    component.find('button#input-submit').simulate('submit')
    expect(sendFn).toHaveBeenCalledWith('hello')
  })
})