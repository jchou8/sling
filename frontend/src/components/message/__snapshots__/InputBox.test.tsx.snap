// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputBox /> should render when disabled 1`] = `ReactWrapper {}`;

exports[`<InputBox /> should render when enabled 1`] = `ReactWrapper {}`;
